packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -133,16 +134,29 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
     }
 
     const decoratorsMeta = getDecoratorsMeta(Ctor);
-    const { apiFields, apiFieldsConfig, apiMethods, wiredFields, wiredMethods, observedFields } =
-        decoratorsMeta;
+    const {
+        apiFields,
+        apiFieldsConfig,
+        apiMethods,
+        wiredFields,
+        wiredMethods,
+        observedFields,
+        trackFields,
+    } = decoratorsMeta;
     const proto = Ctor.prototype;
 
     let { connectedCallback, disconnectedCallback, renderedCallback, errorCallback, render } =
         proto;
     const superProto = getCtorProto(Ctor);
     const superDef =
         superProto !== LightningElement ? getComponentInternalDef(superProto) : lightingElementDef;
-    const bridge = HTMLBridgeElementFactory(superDef.bridge, keys(apiFields), keys(apiMethods));
+    const bridge = HTMLBridgeElementFactory(
+        superDef.bridge,
+        keys(apiFields),
+        getOwnPropertyDescriptors(proto),
```

Comment:
This `getOwnPropertyDescriptors` call is something else that we can potentially avoid in prod mode (as a perf improvement).

Maybe it would be simpler to move your new logic out of `HTMLBridgeElementFactory` entirely? You could then call `defineProperties` yourself without needing to pass anything new into `HTMLBridgeElementFactory`.